core/core.go

```diff
@@ -617,27 +626,32 @@ func (c *Core) readDev(device io.Reader) ([]byte, error) {
 
 func (c *Core) readWriteDev(
 	body []byte,
-	device io.ReadWriter,
+	acquired *session,
 	mode CallMode,
 ) ([]byte, error) {
 
+	acquired.writeMutex.Lock()
```

Comment:
the mutex should only lock if `mode != CallModeRead`, otherwise we are spinning it uselessly.

Reply:
Done.